### ISSUE TITLE
Mission 'Monster Tank Madness': Should mission fail when selling naval yard before having built a transport?

### ISSUE THREAD

**Issue by digitalheir** [OpenRA/OpenRA#16176]
### Issue Summary
The mission cannot proceed of you can't build a transport unit. The mission fails when the naval yard is destroyed, but not when you sell it.
[OpenRA-2019-02-07T072514Z.zip](https://github.com/OpenRA/OpenRA/files/2839854/OpenRA-2019-02-07T072514Z.zip)

... ... ...
- OpenRA Replays
See replay around 2:20

**Comment by digitalheir** [comment]
I was looking around in `monster-tank-madness.lua`. The condition for the mission to fail is:
```
if not beachReached and (USSRSpen.IsDead or Engineer.IsDead) and LstProduced < 1 
```

Definition of `IsDead` in `Actor.cs`:

``` 
public bool IsDead { get { return Disposed || (health != null && health.IsDead); } }
```

I would guess that Dispose() is called when selling a building; is that not the case?!

**Comment by Smittytron** [comment]
Closed by #16177
